src/main/scala/com.mediative.eigenflow/EigenflowBootstrap.scala

```diff
@@ -60,9 +61,16 @@ trait EigenflowBootstrap {
   private def stopSystem(system: ActorSystem)(code: Int): Unit = {
     Try(messagingSystem.stop())
 
-    system.terminate().onComplete {
-      case Success(_) => System.exit(code)
-      case Failure(_) => System.exit(1)
+    // in case if termination process hangs
+    try {
```

Comment:
Can we pinpoint the exception source a bit more? I want to show exactly what we're waiting for that could potentially fail:

```scala
// in case termination process hangs
try { Await.result(system.terminate(), 10.seconds) }
catch {
  case e: java.util.concurrent.TimeoutException =>
    System.err.println("Timeout actor system termination. Forcing system exit.")
    Runtime.getRuntime.halt(1) // force quit with status code 1
}
```